chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/aggregator/workload/MetricLogProcessor.java

```diff
@@ -202,135 +202,99 @@ public StatisticsGenerator generate() throws IOException {
+            long intervalSize = getIntervalSize(intervalSizeProvider, aggregatorSettings, aggregationInfo);
+
+            if (metricDescription.getShowSummary()) {
+                overallMetricAggregator = entry.getKey().provide();
+            }
+            if (metricDescription.getPlotData()) {
+                intervalAggregator = entry.getKey().provide();
+            }
+            if (metricDescription.getShowSummary() || metricDescription.getPlotData()) {
+                MetricAggregator nameAggregator = overallMetricAggregator == null ? intervalAggregator : overallMetricAggregator;
+                String aggregatorName = nameAggregator.getName();
+                String aggregatorIdSuffix = createIdFromName(aggregatorName, normalizeByIntervalValue);
+                String aggregatorDisplayNameSuffix = createAggregatorDisplayNameSuffix(aggregatorName, normalizeByIntervalValue);
+                String displayName = (metricDescription.getDisplayName() == null ? metricDescription.getMetricId() : metricDescription.getDisplayName()) + aggregatorDisplayNameSuffix;
+                String metricId = metricDescription.getMetricId() + '-' + aggregatorIdSuffix;
+                MetricDescriptionEntity metricDesc = persistMetricDescription(metricId, displayName, taskData);
+
+                long currentInterval = aggregationInfo.getMinTime() + intervalSize;
+                long time = intervalSize;
+                long extendedInterval = intervalSize;
+                int addedStatistics = 0;
+                try (LogReader.FileReader<MetricLogEntry> fileReader = logReader.read(path, MetricLogEntry.class)) {
+                    for (MetricLogEntry logEntry : fileReader) {
+                        log.debug("Log entry {} time", logEntry.getTime());
                         if (metricDescription.getPlotData()) {
-                            Number aggregated = intervalAggregator.getAggregated();
-                            if (aggregated != null) {
-                                double value = aggregated.doubleValue();
-
-                                // normalize result
-                                if (normalizeByTimeRequired) {
-                                    long intervalForNormalization;
-                                    if (!normalizeByFullTimeIntervalRequired) {
-                                        // normalize by current aggregation interval
-                                        intervalForNormalization = extendedInterval;
-                                    } else {
-                                        // normalize by time from beginning of the test
-                                        intervalForNormalization = time;
+                            while (logEntry.getTime() > currentInterval) {
+                                Number aggregated = intervalAggregator.getAggregated();
+                                if (aggregated != null) {
```

Comment:
because they were obvious and didn't tell any useful information, which cannot be obtained from the code.